wordpress-mobile/gutenberg-mobile
# added replaceBlockAction handling

Fixes #315 

#### What this PR does
If the currently selected block is empty and the BlockPicker is actioned, the currently selected block is now replaced with the new block type as selected in the BlockPicker.

~There are some cases that I'll want to make an exception for (will check Gutenberg on the web for that), as for example an image block may not have an image set but still we want to keep it?. Will look and make a final description on this PR.~

Only these blocks `core/heading
core/paragraph
core/code
` are taken into account for the condition described above (these are basically blocks where the user is expected to enter some text in, and where it makes sense to check for emptiness). All other blocks will not be replaced.

### ToDo
- [x] replace title `ADD BLOCK` for `REPLACE BLOCK` as corresponds 0a210a9
- [x] check which cases should not be replaced (`core/image` blocks? `core/more` block? etc) d87a8f7  - whitelist so only these blocks `core/heading
core/paragraph
core/code
` are taken into account.

![replaceorinsert](https://user-images.githubusercontent.com/6597771/49530048-26878a80-f896-11e8-9ba7-e35fea21a71a.gif)


#### To test:
1. load the demo app
2. tap on the + inserter icon to bring the BLockType picker up
3. choose Paragraph type (first on the left)
4. observe a new, empty Paragraph block is appended to the list and the editor scrolls there (the Paragraph editor gets the focus and the block is selected)
5. leaving the focus/selection there, tap on the + inserter icon again
6. now select T title block
7. observe the block is now a Title block type, as opposed to appending a new empty block once again.

Repeat the steps 5-7 again with different blocks: when the currently selected block is either of Paragraph, Title, or Code any other selected block in the picker will take its place, whereas if the currently  selected block is any other then whatever pick you make on the Picker will introduce a new block and insert it below.
